util/dvsim/NcLauncher.py

```diff
@@ -0,0 +1,217 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+import datetime
+import os
+import shlex
+import shutil
+import subprocess
+from Launcher import ErrorMessage
+from Launcher import Launcher
+from Launcher import LauncherError
+
+
+class NcLauncher(Launcher):
+    """Implementation of Launcher to launch jobs using altair nc."""
+
+    def __init__(self, deploy):
+        """Initialize common class members."""
+
+        super().__init__(deploy)
+
+        # Popen object when launching the job.
+        self.process = None
+
+    def get_submit_cmd(self, log_file, job_name):
+        cmd_array = []
+        cmd_array.extend(['nc', 'run',
+                          '-e', 'SNAPSHOT',
+                          '-nolog', '-nodb', '-forcelog', '-wl', '-l', log_file,
+                          '-postcmd', '\'',
+                          'nc_post_cmd.sh', '>', 'post.log', ';',
+                          'cat', 'post.log', '>>', log_file, ';',
+                          '\'',
+                          '-set', job_name])
+        return cmd_array
+
+    def create_run_sh(self, full_path, cmd):
+        source_file = os.path.join(os.path.dirname(__file__), 'nc_post_cmd.sh')
+        destination_file = os.path.join(full_path, 'nc_post_cmd.sh')
+        shutil.copy2(source_file, destination_file)
+        run_file = os.path.join(full_path, 'run.sh')
+        try:
+            os.remove(run_file)
+        except FileNotFoundError:
+            pass  # File doesn't exist, which is fine
+        except Exception as e:
+            print(f'Warning: Unable to remove {run_file}: {e}')
+        with open(run_file, 'w') as f:
+            f.write('#!/bin/sh\n')
+            f.write('function realpath {\n')
+            f.write('  python -c "import os; print (os.path.realpath(\'$1\'))"\n')
+            f.write('}\n')
+            f.write('MY_FILEPATH=$(realpath "${BASH_SOURCE[0]}")\n')
+            f.write('MY_DIR=$( dirname "${MY_FILEPATH}" )\n')
+            f.write('cd $MY_DIR\n')
+            f.write('echo Launch start : `date`\n')
+            f.write('SECONDS=0\n')
+            f.write(cmd)
+            f.write('\necho Launch end : `date`')
+            f.write('\necho CPU time : $SECONDS sec')
+        os.chmod(run_file, 0o755)
+
+    def call_submit_cmd(self, exetool, log_file, job_name, cmd, odir):
+        full_path = odir
+        cmd_array = self.get_submit_cmd(log_file, job_name)
+        self.create_run_sh(full_path, cmd)
+        # TODO: This needs to be moved to a HJson.
+        if (exetool == 'xcelium'):
+            cmd_array.extend(['-r', 'License:Xcelium_Single_Core/1'])
+        if (exetool == 'vcs'):
+            cmd_array.extend(['-r', 'License:VCSRuntime_Net/1'])
```

Comment:
I was trying to keep the implementation similar to the one here 
https://github.com/lowRISC/opentitan/blob/master/util/dvsim/LsfLauncher.py#L183
For example, we can also fork verilator runs using nc tool , but because its license free, we dont need any rusage(-r) flags

Reply:
Thanks for explaining: that makes sense to me!